File: BubbleSort.cpp
/*
 * Bubble sort
 */

#include<iostream>
using namespace std;

class BubbleSort{

	public:
	
	void swap(int *x, int *y){
		int temp = *x;
		*x = *y;
		*y = temp; 
	}
	
	
	void printArray(int array[], int length){
		for(int i = 0; i < length; i++){
			cout << array[i] << "\t";
		}
	}
	
	void sort(int array[], int length){
		
		int temp;
		
		for(int i=length-1; i>=0; i--){
			
			for(int j=i; j>=0; j--){
				
				if(array[j] < array[j+1]){
					swap(&array[j], &array[j+1]);
				}
			}
		}
		
		printArray(array, length);
	}
};

int main(){
	
	int array[] = {64, 34, 25, 12, 22, 11, 90};
	int length = sizeof(array) / sizeof(array[0]);
	
	BubbleSort obj;
	obj.sort(array, length);
	
	return 0;
}

File: BubbleSort.java
import java.util.Scanner;
public class BubbleSort {
	
	public void sort(int arr[]){
		
		for(int i=0; i<arr.length-1; i++){
			for(int j=0; j<arr.length-i-1; j++){
				if(arr[j] > arr[j+1]){
					int temp = arr[j];
					arr[j] = arr[j+1];
					arr[j+1] = temp;
				}
				printArray(arr);
			}
		}
	}
	
	public void printArray(int arr[]){
		for(int i=0; i<arr.length; i++){
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}
	
	public static void main(String...strings){
		
		BubbleSort obj = new BubbleSort();
		Scanner in = new Scanner(System.in);
		
		int size = in.nextInt();
		int[] arr = new int[size];
		
		for(int i=0; i<size; i++){
			arr[i] = in.nextInt();
		}
		obj.sort(arr);
		in.close();
	}
}
